### DISCUSSION TITLE
Create XBRL document

### DISCUSSION THREAD

**Discussion by prasad-soulpage** [manusimidt/py-xbrl#116] [General]
Just wondering if this API can be used to create a fresh XBRL document from scratch. If not, could you please point me to any open source resources ?

**Comment by manusimidt** [comment]
No, this library is purely designed for data retrieval and not for creating and/or validating XBRL documents. 

Unfortunately, the OpenSource XBRL landscape is pretty narrow. Most applications are commercial and closed-source. I would recommend to look into [Arelle](https://github.com/Arelle/Arelle). However, I am not really familiar with all the functionalities of Arelle, so I can't give you more detailed advice.
